buildSrc/src/main/kotlin/elastic-otel.sign-and-publish-conventions.gradle.kts

```diff
@@ -0,0 +1,91 @@
+plugins {
+  `maven-publish`
+  publishing
+  signing
+}
+
+interface PublishingConventionsPluginExtension {
+  /**
+   * By default the convention will publish the artifacts and pom as libraries.
+   * To override the behaviour provide the tasks producing the artifacts as this property.
+   * This should only be required when publishing fat-jars with custom packaging.
+   */
+  val artifactTasks: ListProperty<Task>
+}
+
+val publishingConventions = project.extensions.create<PublishingConventionsPluginExtension>("publishingConventions")
+publishingConventions.artifactTasks.convention(listOf())
+
+
+afterEvaluate {
+
+  publishing {
+
+    repositories {
+      // dry-run repository will be wiped on 'clean' task
+      maven {
+        name = "dryRun"
+        url = rootProject.layout.buildDirectory.dir("dry-run-maven-repo").get().asFile.toURI()
+      }
+    }
+
+    publications {
+      register("maven", MavenPublication::class) {
+        artifactId = base.archivesName.get()
+
+        val artifactTasks = publishingConventions.artifactTasks.get()
+        if (artifactTasks.isEmpty()) {
+          //publish as library with dependencies
+          from(components["java"])
+          versionMapping {
+            usage("java-api") {
+              fromResolutionOf("runtimeClasspath")
+            }
+            usage("java-runtime") {
+              fromResolutionResult()
+            }
+          }
+        } else {
+          //publish just the artifacts (fat-jars).
+          for (task in artifactTasks) {
+            artifact(task)
+          }
+        }
+
+        pom {
+          name.set(project.description)
+          description.set(project.description)
```

Comment:
[minor] Maybe that would be worth a later improvement but do we have default values for the project description ? Or maybe it's probably better to make this mandatory and throw an explicit error when value is not provided.

Reply:
Good point, I'll add a check to throw an error if the description is unset.

Reply:
If the description is missing, gradle will now show an error at configuration time (so even without trying to publish).